### ISSUE TITLE
Shift+F4 Editor should open filenames like filename.cpp:123 or filename:123:34 with reposition of cursor.

### ISSUE THREAD

**Issue by alexandrustinov** [elfmz/far2l#1476]
1) Many of programmers tools (gcc/clang/addr2line/etc/so on) output filenames with line and column number suffix, in form

    filename:nnn
    filename:nnn:ccc

It would be great if it will be possible to open file pasting next string in Shift+F4 dialog:
    /home/user/Projects/system/far2l/far2l/far2l/src/console/keyboard.cpp:996

that way Far2L could open file /home/user/Projects/system/far2l/far2l/far2l/src/console/keyboard.cpp and move cursor to the line 996

and if I put filename ../poc/src/nett.cpp:44:4 then it should open file ../poc/src/nett.cpp and then locate to line 44 and column 4.

2) The same might be applicable to command line as well, like:

    <pre><code>far2ledit filename[:line[:pos]]</pre></code>

in addition to currently implemented form:

     -e[line[:pos]] filename
          Edit the specified file with optional cursor position specification.

Reason: Midnight Commander has utility mcedit (please see <code>man mcedit</code> ) - that way is possible to replace standard EDITOR and forget about vi.

But Far2L does not provide binary which allows to execute command like <code>mcedit filename:123</code>, it's a bit tricky to replace the system EDITOR by far2l 
It's possible using bash script, but it will require some additional efforts to split the filename[:nnn[:ccc]] parameter.

**Comment by elfmz** [comment]
not too much tricky :)
```
echo "$1" | awk -F':' '{ \
	if ($3 != "") {system("far2l -e" $2 ":" $3 " \"" $1 "\""); } \
	else if ($2 != "") {system("far2l -e" $2 " \"" $1 "\""); } \
	else {system("far2l -e \"" $1 "\""); } \
} '

```
note that : is a valid name part so such change will  prevent editing files named foo:bar ...

**Comment by alexandrustinov** [comment]
Yes, it's not rocket science, but each user will need to do that by his own, just to see far2l editor after <code>crontab -e</code>
Also Far2L could set EDITOR variable for if it is not explicitly defined. I suppose that lot of the user still don't know that it is even possible to get rid of vi that way.

**Comment by alexandrustinov** [comment]
> note that : is a valid name part so such change will prevent editing files named foo:bar ...

That is true, mcedit is able to handle that

<code>echo 123 > test:12</code>
<code>mcedit test:12</code>

<code>echo 222 > test</code>
<code>mcedit test:1</code>

<code>mcedit test:12:1</code>

everything works fine there.

Meanwhile mcedit is not able to set column position, command mcedit test:1:2 will lead to nowhere, unfortunately, that was most annoying thing there.

**Comment by alexandrustinov** [comment]
Visual Studio Code has such feature.

It's possible to open Terminal inside the IDE, than perform

    echo test:4:3

then hover mouse cursor over text <code>test:4:3</code> and press Ctrl+Click: vscode will open file <code>test</code> and navigate to line 4 and column 3 that way - it's very nice feature when you are fixing bugs/warnings/backtraces/etc

But for text <code>test:12</code> and even <code>test:12:1</code> from sample above it will not work: vscode tries to open file <code>test</code> and navigate to line 12.
Though clicking on text <code>test:12:1:1</code> will finally try to open file <code>test:12</code> at line 1 and column 1.

So correct <filename:n:m> resolution logic might sound like below:
try to open file <code>filename:n:m</code> if it exists 
or try to open file <code>filename:n</code> (and navigate to line m) if it exists 
or try to open file <code>filename</code> and navigate to line n and column m of file exist, 
or just create new <filename:n:m> if there were no existing files - as far as n:m does not have sense for new files.

Suffixes have meaning for existing file names,

**Comment by alexandrustinov** [comment]
mmmm.... sorry, the Far2L output screen shown by Ctrl+O.

It seems that Visual Studio Code like trick with Ctrl+Click on displayed filename does not work there.

Far2L could act just like vscode terminal - open path/filename, path/filename:line, path/filename:line:column if the file with filename is present in path starting from current directory.

Reason is the same - that way you can compile code, cmake/make/gcc will output filenames in filename:line:column form but you will be able to open the problem lines in files just by Ctrl+click right over text in the terminal window (or/and related editor). No need to stuck with Alt+Ins and so on form of filename selection, Copy, Shift+F4, Paste, Enter... navigate line 8192...
